src/app/_mermaid_common/libs/factories/offline-table.factory.js

```diff
@@ -431,23 +437,33 @@ angular.module('mermaid.libs').factory('OfflineTable', [
           var relatedColumns = joinSchema.relatedColumns;
           var foreignKey = joinSchema.foreignKey;
           var relatedKey = joinSchema.relatedKey;
-
-          var noMatchRecord = _.zipObject(
-            relatedColumns,
-            _.map(relatedColumns, function() {
-              return null;
-            })
-          );
+          var many = joinSchema.many || false;
 
           where[relatedKey] = null;
           return _.map(records, function(obj) {
             obj[tablePrefix] = obj[tablePrefix] || {};
-            where[relatedKey] = _.get(obj, foreignKey);
-            var match = _.find(relatedRecords, where);
 
-            if (match === undefined) {
-              obj[tablePrefix] = noMatchRecord;
+            if (_.isFunction(joinSchema.relateFunction)) {
+              where = function(relatedRecord) {
+                return joinSchema.relateFunction(
+                  obj,
+                  relatedRecord,
+                  joinSchema
+                );
+              };
+            } else {
+              where[relatedKey] = _.get(obj, foreignKey);
+            }
+
+            var matches = _.filter(relatedRecords, where);
+
+            obj[tablePrefix] = null;
+            if (many === true) {
```

Comment:
`filter`/`many` is much better, but we're not actually using it anywhere yet, right? Good for future?

Reply:
was going to initially use it but changed plan, but thought it would be useful so I kept it around